packages/webapp/src/components/Flexbox/Flexbox.tsx

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { FC } from 'react'
+import React, { memo, useMemo } from 'react'
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+interface IFlex<T = {}> extends FC<T> {
+	Box: FC<BoxProps>
+}
+
+export interface FlexProps {
+	as?: React.ElementType
+	vertical?: boolean
+	wrap?: boolean
+	justify?:
+		| 'flex-start'
+		| 'flex-end'
+		| 'center'
+		| 'space-between'
+		| 'space-around'
+		| 'space-evenly'
+	align?: 'stretch' | 'flex-start' | 'flex-end' | 'center' | 'baseline'
+	gap?: string
+	children?: React.ReactNode
+	className?: string
+	style?: React.CSSProperties
+}
+
+// @ts-expect-error Doesn't meet type constraint until later
+export const Flex: IFlex<FlexProps> = memo(function Flex({
```

Comment:
Maybe we should move this later on to a utility library. I can see this being useful to other projects.